### DISCUSSION TITLE
React 18 streaming support

### DISCUSSION THREAD

**Discussion by HarshwardhanSingh** [vikejs/vike#1276] [Discussion] [answered]
First of all, thanks for this amazing piece of software, I was trying to have the react streaming  integration using `renderToPipableStream` with suspense for data  fetching and could not find exact instructions on how to achieve that. I have referred to the examples of react integration but still stuck with the integration. I can share more details but wanted to ask if there is already a setup for this config.

If I am able to figure out in the meantime I will share a working demo for others to refer.

Thanks

**Comment by brillout** [comment]
I'm glad you like it :). Minimal reproduction welcome and we'll have a look at it.

**Comment by lourot** [comment, marked as answer]
@HarshwardhanSingh the examples make use of https://github.com/brillout/react-streaming, which uses `renderToPipableStream()` under the hood.

Also I'm adding another example in #1271, but while doing that I discovered that [progressive/selective hydration](https://github.com/reactwg/react-18/discussions/37) was broken and I'll work on a fix in a couple of days. But this shouldn't block you, it's just that you'll get a performance boost when this is fixed.

If you like using Vike, have you or your company considered [sponsoring](https://github.com/sponsors/vikejs/)? It makes a big difference for the project and you'll get a priority boost on your issues

**Reply by HarshwardhanSingh** [reply]
Thanks a lot @AurelienLourot 
I was able to get it working using  ```react-streaming```. I should have looked deeper 🙂.

**Reply by lourot** [reply]
Happy you got it working :) And thanks again for having started sponsoring us @HarshwardhanSingh (https://github.com/brillout/docpress/pull/5)

**Reply by brillout** [reply]
> Quick question: Is it possible to make this work with react-query out of the box, since the react-query has its own alternative for useAsync

There is actually work-in-progress about exactly that: https://github.com/vikejs/vike/issues/1263. Also have a look at [Telefunc](https://telefunc.com/) in case you own the backend.
